### DISCUSSION TITLE
restore a ful backup.gz file

### DISCUSSION THREAD

**Discussion by thierry-carron** [supabase/supabase#26664] [Questions]
Hi there, (I am new)
I don't see how to select a local backup.gz file to full reload into new blank project ?https://supabase.com/dashboard/project/fxxxxxxxxxxxxxxxxxxxxxxxxxxxx/database/backups/scheduled

**Comment by thierry-carron** [comment]
nobody to help here ????

**Comment by brunowego** [comment]
Follow the steps below, you will see that the `.backup` file is an SQL file.

## Download Backup

1. [Supabase Projects](https://supabase.com/dashboard/projects)
2. Select a project
3. Mein Menu -> Database
4. Aside Menu -> Platform -> Backups
5. Database Backups -> Scheduled backups (Tab) -> Select one -> Download

## Restore Backup (Local)

```sh
#
gunzip <filename>.backup.gz

#
file --mime-type <filename>.backup

#
head -n 20 <filename>.backup
```

<img width="803" alt="image" src="https://github.com/user-attachments/assets/0aafd794-6db0-4ecc-8206-e188075a1f4c">

**Reply by mauriciopiber** [reply]
Great solution, worked like a charm.

**Reply by brunowego** [reply]
Below is a tip on how to import locally:

```sh
#
sed -i 's/^\(\\connect postgres\)$/\\connect acme/' <filename>.backup

#
export DATABASE_URL='postgresql://acme:acme@localhost:5432/acme'

#
psql -d "$DATABASE_URL" -c 'ALTER USER acme WITH SUPERUSER'

#
psql -d "$DATABASE_URL" < ./<filename>.backup
```

The default Supabase database name is `postgres`, `sed` is for you to change it to another name.

**Reply by brunowego** [reply]
It is normal to show some errors in the output, you can ignore them... and wait until it finishes.

**Reply by mauriciopiber** [reply]
@brunowego yep with all those tips it become easy.

I just had to create the user, password and database locally + your code and everything worked.

Thanks! You're the best.

**Comment by MechaMahmud71** [comment]
For docker container this can be used.
```
sed -i 's/^\(\\connect postgres\)$/\\connect <name_of_the_db>/' <backup_file_name>.backup
docker cp <backup_file_name>.backup <postgres_container_name>:/<backup_file_name>.backup
docker exec -e PGPASSWORD=<db_password> <pastgres_container_name> psql -U <db_username> -d <db_name> -c "ALTER USER <db_user_name> WITH SUPERUSER"
docker exec -t <postgres_conatiner_name> bash
psql -U <user_name> -d <db_name> -f /<backup_file_name>.backup
```
